bgpd/bgpd.c

```diff
@@ -4902,6 +4916,43 @@ int peer_ebgp_multihop_unset(struct peer *peer)
 	return 0;
 }
 
+/* Set Open Policy Role and check its correctness */
+int peer_role_set(struct peer *peer, uint8_t role, int strict_mode)
+{
+	if (peer->local_role == role) {
+		if (CHECK_FLAG(peer->flags, PEER_FLAG_STRICT_MODE) &&
+		    !strict_mode)
+			/* TODO: Is session restart needed if it was down? */
+			UNSET_FLAG(peer->flags, PEER_FLAG_STRICT_MODE);
```

Comment:
I suggest using peer_flag_set_vty()/unset. It takes the action to do none/restart/in/out according to the flag.

Reply:
Do we really need a full reset? I'm talking about the cases where graceful-restart is enabled and notification support for GR is enabled too (to retain the routes on CEASE/Config changed or so).

Reply:
IMO, let's try this approach as it is.